test/Constraints/dynamic_lookup.swift

```diff
@@ -210,7 +210,6 @@ type(of: obj).foo!(obj)(5) // expected-error{{instance member 'foo' cannot be us
 
 // Checked casts to AnyObject
 var p: P = Y()
-// expected-warning @+1 {{forced cast from 'P' to 'AnyObject' always succeeds; did you mean to use 'as'?}}
```

Comment:
It's diagnosed by CSApply so that's why it's lost when error is diagnosed, but it's not really lost-lost, when problem with `!` is fixed, that warning is going to be displayed again.

Reply:
Let me add one more test here to illustrate that...

Reply:
I think I can make it so this warning is produced as a fix too, that's on my plate. It would have to be done separately though. The problem here is that our diagnostics are scattered all over the place at the moment, and _by chance_ in this case sub-expression type-checks and produces this warning.